src/ripple/app/misc/TxQ.h

```diff
@@ -149,9 +158,19 @@ class TxQ
         amendment is not enabled, OR if the account has no transactions
         in the queue.
     */
-    boost::optional<std::map<TxSeq, AccountTxDetails>>
+    boost::optional<std::map<TxSeq, AccountTxDetails const>>
     getAccountTxs(AccountID const& account, ReadView const& view) const;
 
+    /** Returns information about all transactions currently
+        in the queue.
+
+        @returns Uninitialized @ref optional if the FeeEscalation
+        amendment is not enabled, OR if there are no transactions
+        in the queue.
+    */
+    boost::optional<std::vector<TxDetails>>
```

Comment:
Admittedly not the strongest defense but I'll state a case.
- Avoids instantiating an empty vector in cases where it is not necessary.
- Convenient when calling the LedgerFill constructor.
- The semantics can be cleaner e.g. ` if (auto txs = getTxs(...))` vs calling `txs.empty()`.
- It is kewl :)